### ISSUE TITLE
[BUG] Springboot application failing to start due to AppConfigurationRefreshEndpoint final class from azure appconfiguration-web with actuator dependency

### ISSUE THREAD

**Issue by SriPalla** [Azure/AppConfiguration#917]
**Describe the bug**
Springboot application failing to start due to AppConfigurationRefreshEndpoint final class from azure appconfiguraiton-web when actuator starter is being added to project.

**Exception or Stack Trace**

2024-05-13 20:46:10,545 INFO - [main] com.demo.DemoApplicationKt:660 : The following 1 profile is active: "local"
2024-05-13 20:46:11,299 WARN - [main] org.springframework.boot.actuate.endpoint.EndpointId:155 : Endpoint ID 'appconfiguration-refresh' contains invalid characters, please migrate to a valid format.
2024-05-13 20:46:11,649 WARN - [main] org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext:632 : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'appConfigurationRefreshEndpoint' defined in class path resource [com/azure/spring/cloud/appconfiguration/config/web/implementation/AppConfigurationWebAutoConfiguration$AppConfigurationPushRefreshConfiguration.class]: Could not generate CGLIB subclass of class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint: Common causes of this problem include using a final class or a non-visible class
2024-05-13 20:46:11,671 ERROR - [main] org.springframework.boot.SpringApplication:851 : Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'appConfigurationRefreshEndpoint' defined in class path resource [com/azure/spring/cloud/appconfiguration/config/web/implementation/AppConfigurationWebAutoConfiguration$AppConfigurationPushRefreshConfiguration.class]: Could not generate CGLIB subclass of class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint: Common causes of this problem include using a final class or a non-visible class
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:607)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522)
at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:325)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:323)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975)
at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:959)
at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:624)
at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146)
at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:334)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:1354)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
at com.demo.DemoApplicationKt.main(DemoApplication.kt:20)
Caused by: org.springframework.aop.framework.AopConfigException: Could not generate CGLIB subclass of class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint: Common causes of this problem include using a final class or a non-visible class
at org.springframework.aop.framework.CglibAopProxy.buildProxy(CglibAopProxy.java:227)
at org.springframework.aop.framework.CglibAopProxy.getProxy(CglibAopProxy.java:160)
at org.springframework.aop.framework.ProxyFactory.getProxy(ProxyFactory.java:110)
at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.buildProxy(AbstractAutoProxyCreator.java:517)
at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.createProxy(AbstractAutoProxyCreator.java:464)
at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.wrapIfNecessary(AbstractAutoProxyCreator.java:369)
at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.postProcessAfterInitialization(AbstractAutoProxyCreator.java:318)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:438)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1789)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:600)
... 15 more
Caused by: java.lang.IllegalArgumentException: Cannot subclass final class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint
at org.springframework.cglib.proxy.Enhancer.generateClass(Enhancer.java:653)
at org.springframework.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:26)
at org.springframework.cglib.core.ClassLoaderAwareGeneratorStrategy.generate(ClassLoaderAwareGeneratorStrategy.java:57)
at org.springframework.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:366)
at org.springframework.cglib.proxy.Enhancer.generate(Enhancer.java:575)
at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData.lambda$new$1(AbstractClassGenerator.java:107)
at org.springframework.cglib.core.internal.LoadingCache.lambda$createEntry$1(LoadingCache.java:52)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at org.springframework.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:57)
at org.springframework.cglib.core.internal.LoadingCache.get(LoadingCache.java:34)
at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:130)
at org.springframework.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:317)
at org.springframework.cglib.proxy.Enhancer.createHelper(Enhancer.java:562)
at org.springframework.cglib.proxy.Enhancer.createClass(Enhancer.java:407)
at org.springframework.aop.framework.ObjenesisCglibAopProxy.createProxyClassAndInstance(ObjenesisCglibAopProxy.java:62)
at org.springframework.aop.framework.CglibAopProxy.buildProxy(CglibAopProxy.java:218)
... 24 more

Process finished with exit code 1

**To Reproduce**
Adding Springboot actuator along with azure dependencies as listed below
implementation("com.azure.spring:spring-cloud-azure-starter-appconfiguration-config-web")
implementation("com.azure.spring:spring-cloud-azure-feature-management-web")
implementation("org.springframework.boot:spring-boot-starter-actuator")
mavenBom("org.springframework.cloud:spring-cloud-dependencies:2023.0.1")
plugins {
id("org.springframework.boot") version "3.2.3"
}
Code Snippet

Expected behavior
Application should start successfully

**Comment by SriPalla** [comment]
@mrm9084 Springboot version: 3.2.3 spring-aop: 6.1.4

**Comment by jimmyca15** [comment]
> It seems like it is an issue with Spring AOP and our library

@mrm9084 were you able to confirm this?

**Comment by mrm9084** [comment]
So where I've seen this issue before is when Spring Native gets involved and it tries to Mock out one of our classes that's marked as final. I'm trying to repo, but haven't had any luck yet.

I'm guessing part of it might be our setup to mainly work with just Spring Boot, or Spring Boot + Spring Web, but not any of the other Spring frameworks like Web.

**Comment by mrm9084** [comment]
@SriPalla I'm having trouble replicating this, do you have a sample that replicates the issue. Also, the bug states that the application fails to start, but the top of the stack trace says

```
org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext:632 : Exception encountered during context initialization - cancelling refresh attempt: 
```

Which would only happen if the original started happened.

**Comment by SriPalla** [comment]
Hi @mrm9084 
Gradle kotlin file with dependencies:
-------------------------------------------
```
import com.google.protobuf.gradle.id
import com.netflix.graphql.dgs.codegen.gradle.GenerateJavaTask

import org.jetbrains.kotlin.gradle.dsl.JvmTarget
import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
import org.springframework.boot.gradle.tasks.bundling.BootJar


plugins {
    id("org.springframework.boot") version "3.2.5"
    id("io.spring.dependency-management") version "1.1.5"
    id("io.gitlab.arturbosch.detekt") version "1.23.6"
    id("org.sonarqube") version "4.3.1.3277"
    id("com.netflix.dgs.codegen") version "6.1.4"
    id("com.google.protobuf") version "0.9.4"
    id("org.openapi.generator") version "7.4.0"
    kotlin("jvm") version "1.9.24"
    kotlin("plugin.spring") version "1.9.24"
    jacoco
}

group = "com.demo"
version = "0.0.1-SNAPSHOT"
java.sourceCompatibility = JavaVersion.VERSION_17



configurations.all {
    exclude(group = "org.springframework.boot", module = "spring-boot-starter-logging")
}



val dgsVersion = "8.2.4"
val eventHubVersion = "5.18.1"
val googleProtocVersion = "3.25.3"
val grpcProtobufKotlinVersion = "1.4.1"
val grpcProtobufVersion = "1.64.0"
val jacocoVersion = "0.8.12"
val karateJunit5Version = "1.5.0.RC4"
val ktLintVersion = "0.50.0"
val lmaxDisruptorVersion = "3.4.4"
val masterThoughtVersion = "5.7.8"
val mockkVersion = "1.13.11"
val splunkVersion = "1.30.0"
val springdocVersion = "2.3.0"
val dataFakerVersion = "2.0.2"
val mockitoKotlinVersion = "5.2.1"

extra["kotlin.version"] = "1.9.24"
extra["kotlin-coroutines.version"] = "1.8.1"
extra["springCloudVersion"] = "2023.0.1"
extra["springCloudAzureVersion"] = "5.12.0"

dependencies {
    implementation("org.springframework.boot:spring-boot-starter-web")
    implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
    implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310")
    implementation("org.springframework.boot:spring-boot-starter-aop")
    implementation("org.springframework.boot:spring-boot-actuator")
    implementation("org.jetbrains.kotlin:kotlin-stdlib")
    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core")

    implementation("com.netflix.graphql.dgs:graphql-dgs-spring-boot-starter")
    implementation("com.netflix.graphql.dgs:graphql-dgs-extended-scalars")

    implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:$springdocVersion")

    implementation("io.grpc:grpc-protobuf:$grpcProtobufVersion")
    implementation("io.grpc:grpc-stub:$grpcProtobufVersion")
    implementation("io.grpc:grpc-netty:$grpcProtobufVersion")
    implementation("com.google.protobuf:protobuf-java-util:$googleProtocVersion")
    implementation("com.google.protobuf:protobuf-kotlin:$googleProtocVersion")
    implementation("io.grpc:grpc-kotlin-stub:$grpcProtobufKotlinVersion")
    implementation("io.grpc:protoc-gen-grpc-kotlin:$grpcProtobufKotlinVersion")

    implementation("org.springframework.boot:spring-boot-starter-log4j2")
    implementation("org.apache.logging.log4j:log4j-layout-template-json")
    implementation("com.lmax:disruptor:$lmaxDisruptorVersion")
    implementation("com.splunk:splunk-otel-javaagent:$splunkVersion")

    implementation("com.azure:azure-messaging-eventhubs:$eventHubVersion")
    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-slf4j")
    implementation("io.mockk:mockk:$mockkVersion")


    implementation("com.azure.spring:spring-cloud-azure-appconfiguration-config-web") {
        exclude("org.springframework.security", "spring-security-rsa")
    }
    implementation("org.springframework.security:spring-security-rsa:1.1.3")

    implementation("com.azure.spring:spring-cloud-azure-feature-management-web")
    implementation("org.springframework.boot:spring-boot-starter-actuator")

    testImplementation("org.springframework.boot:spring-boot-starter-test") {
        exclude("org.xmlunit", "xmlunit-core")
    }
    testImplementation("org.xmlunit:xmlunit-core:2.10.0")
    testImplementation("io.karatelabs:karate-junit5:$karateJunit5Version")

    testImplementation("io.grpc:grpc-testing:$grpcProtobufVersion")
    testImplementation("io.grpc:grpc-inprocess:$grpcProtobufVersion")
    testImplementation("ch.qos.logback:logback-classic")
    testImplementation("ch.qos.logback:logback-core")
    testImplementation("net.masterthought:cucumber-reporting:$masterThoughtVersion")
    testImplementation("net.datafaker:datafaker:$dataFakerVersion")
    testImplementation("org.mockito.kotlin:mockito-kotlin:$mockitoKotlinVersion")
    testImplementation("io.jsonwebtoken:jjwt:0.12.5")
    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
}

dependencyManagement {
    imports {
        mavenBom("com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:$dgsVersion")
        mavenBom("org.springframework.cloud:spring-cloud-dependencies:${property("springCloudVersion")}")
        mavenBom("com.azure.spring:spring-cloud-azure-dependencies:${property("springCloudAzureVersion")}")
    }
}
```
------------
springboot bootstrap.yaml file
```
spring:
  cloud:
    azure:
      appconfiguration:
        enabled:  false
        stores[0]:
          connection-string: <endpoint>
          fail-fast: true
          monitoring:
            enabled: true
            refresh-interval: 5s
            triggers[0]:
              key: sentinel
          feature-flags:
            enabled: true
```
----------------------
if spring.cloud.azure.appconfiguration.enabled is `false` everything works with the actuator dependency in place.
in case of spring.cloud.azure.appconfiguration.enabled is `true` then application fails to start.

Startup Logs
```

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v3.2.5)

2024-05-29 18:25:08,270 WARN   - [main] com.azure.core.http.netty.implementation.Utility:293 : The following Netty dependencies have versions that do not match the versions specified in the azure-core-http-netty pom.xml file. This may result in unexpected behavior. If your application runs without issue this message can be ignored, otherwise please update the Netty dependencies to match the versions specified in the pom.xml file. Versions found in runtime: 'io.netty:netty-common' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-handler' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-handler-proxy' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-buffer' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-codec' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-codec-http' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-codec-http2' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-transport-native-unix-common' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-transport-native-epoll' version: 4.1.109.Final (expected: 4.1.101.Final),'io.netty:netty-transport-native-kqueue' version: 4.1.109.Final (expected: 4.1.101.Final)
2024-05-29 18:25:09,497 INFO   - [main] com.demo.demomicroservice.DemoMicroserviceApplicationKt:660 : The following 1 profile is active: "local"
2024-05-29 18:25:10,320 WARN   - [main] org.springframework.boot.actuate.endpoint.EndpointId:155 : Endpoint ID 'appconfiguration-refresh' contains invalid characters, please migrate to a valid format.
2024-05-29 18:25:10,666 WARN   - [main] org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext:632 : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'appConfigurationRefreshEndpoint' defined in class path resource [com/azure/spring/cloud/appconfiguration/config/web/implementation/AppConfigurationWebAutoConfiguration$AppConfigurationPushRefreshConfiguration.class]: Could not generate CGLIB subclass of class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint: Common causes of this problem include using a final class or a non-visible class
2024-05-29 18:25:10,688 ERROR  - [main] org.springframework.boot.SpringApplication:851 : Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'appConfigurationRefreshEndpoint' defined in class path resource [com/azure/spring/cloud/appconfiguration/config/web/implementation/AppConfigurationWebAutoConfiguration$AppConfigurationPushRefreshConfiguration.class]: Could not generate CGLIB subclass of class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint: Common causes of this problem include using a final class or a non-visible class
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:607)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522)
    at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326)
    at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
    at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324)
    at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975)
    at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:962)
    at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:624)
    at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146)
    at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
    at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:334)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1354)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
    at com.demo.demomicroservice.DemoMicroserviceApplicationKt.main(DemoMicroserviceApplication.kt:20)
Caused by: org.springframework.aop.framework.AopConfigException: Could not generate CGLIB subclass of class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint: Common causes of this problem include using a final class or a non-visible class
    at org.springframework.aop.framework.CglibAopProxy.buildProxy(CglibAopProxy.java:230)
    at org.springframework.aop.framework.CglibAopProxy.getProxy(CglibAopProxy.java:163)
    at org.springframework.aop.framework.ProxyFactory.getProxy(ProxyFactory.java:110)
    at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.buildProxy(AbstractAutoProxyCreator.java:519)
    at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.createProxy(AbstractAutoProxyCreator.java:466)
    at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.wrapIfNecessary(AbstractAutoProxyCreator.java:371)
    at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.postProcessAfterInitialization(AbstractAutoProxyCreator.java:320)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:438)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1789)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:600)
    ... 15 more
Caused by: java.lang.IllegalArgumentException: Cannot subclass final class com.azure.spring.cloud.appconfiguration.config.web.implementation.pushrefresh.AppConfigurationRefreshEndpoint
    at org.springframework.cglib.proxy.Enhancer.generateClass(Enhancer.java:653)
    at org.springframework.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:26)
    at org.springframework.cglib.core.ClassLoaderAwareGeneratorStrategy.generate(ClassLoaderAwareGeneratorStrategy.java:57)
    at org.springframework.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:366)
    at org.springframework.cglib.proxy.Enhancer.generate(Enhancer.java:575)
    at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData.lambda$new$1(AbstractClassGenerator.java:107)
    at org.springframework.cglib.core.internal.LoadingCache.lambda$createEntry$1(LoadingCache.java:52)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at org.springframework.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:57)
    at org.springframework.cglib.core.internal.LoadingCache.get(LoadingCache.java:34)
    at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:130)
    at org.springframework.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:317)
    at org.springframework.cglib.proxy.Enhancer.createHelper(Enhancer.java:562)
    at org.springframework.cglib.proxy.Enhancer.createClass(Enhancer.java:407)
    at org.springframework.aop.framework.ObjenesisCglibAopProxy.createProxyClassAndInstance(ObjenesisCglibAopProxy.java:62)
    at org.springframework.aop.framework.CglibAopProxy.buildProxy(CglibAopProxy.java:221)
    ... 24 more

Process finished with exit code 1

```

**Comment by mrm9084** [comment]
@SriPalla, you you take that setup, minus dgs (this shouldn't matter), with minimal app:

```java
package com.example.demo;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class DemoApplication {

	public static void main(String[] args) {
		SpringApplication.run(DemoApplication.class, args);
	}

}
```

It starts up fine, with App Configuration enabled. I even added an endpoint to trigger refresh and it works fine. Are you doing something extra besides just running the Gradle build?

**Comment by mrm9084** [comment]
@SriPalla. Thanks. Was able to replicate. Fix is in PR. Workaround for now.

Instead use:

`com.azure.spring:spring-cloud-azure-appconfiguration-config` not starter or web version. This version was created for just Spring Boot libraries and doesn't have the auto refresh feature built in. If you require the refresh feature you can use it but manually. See: https://learn.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-java-spring-app?tabs=spring-boot-3#use-manual-refresh.

**Comment by zhenlan** [comment]
@mrm9084 can this issue be closed?

**Comment by mrm9084** [comment]
This should be fixed in 5.13.0 or newer.
